Ub fix bandersnatch provider

pcseanmckay opened the pull request:
Updated bandersnatch provider to better parse the provider.conf config file.

Commit:
Updated bandersnatch provider.

Updated bandersnatch provider to replace the directory parameter with
the mirror_tld/repo_name directory being used for the sync.

Commit:
Updated provider_wrapper script to use regex.

Updated the wrapper script to use a regex pattern match to look for the
directory parameter.

codrcodz reviewed (changes requested):
Same feedback as #6 

- Avoid calling external binaries when provisions already exist in bash for the thing you are trying to do
- Should we use sed instead of appending to the end of the file, or does location of not matter? What happens if two lines for the same parameter get dropped in the same file? does it accept the final one?

pcseanmckay commented:
See my relevant comments in PR #6.  Once again if you agree, then I will make the necessary changes.

pcseanmckay committed:
Updated bandersnatch provier_wrapper.sh.

Updated the provider_wrapper.sh script with updates from the apt-mirror
provider_wrapper.sh PR.

codrcodz commented on providers.d/bandersnatch/provider_wrapper.sh:
```diff
@@ -12,16 +12,35 @@ bashelliteProviderWrapperBandersnatch() {
+    else
+      echo "${line}" >> ${tmp_config_file}
+    fi
+    line_counter=$(($line_counter + 1))
```
This line needs to be inside the while loop near your `echo "directory` line. Right now, if the first line in the config file is something besides `^directory =`, your line_counter variable will get incremented, and your line_counter test will always hit your `else` conditional for each subsequent line.

codrcodz reviewed (changes requested):
Correct logic issue.

Commit:
Updated output messages to use provider variable and fixed counter error.
